packages/twenty-front/src/modules/analytics/hooks/useEventTracker.ts

```diff
@@ -1,32 +1,44 @@
 import { useCallback } from 'react';
-import { useRecoilValue } from 'recoil';
-
-import { telemetryState } from '@/client-config/states/telemetryState';
 import { useTrackMutation } from '~/generated/graphql';
-
-interface EventLocation {
+export interface EventData {
   pathname: string;
+  userAgent: string;
+  timeZone: string;
+  locale: string;
+  href: string;
+  referrer: string;
 }
 
-export interface EventData {
-  location: EventLocation;
-}
+export const getSessionId = (cookieName: string): string => {
+  const cookie: { [key: string]: string } = {};
+  document.cookie.split(';').forEach((el) => {
+    const [key, value] = el.split('=');
+    cookie[key.trim()] = value;
+  });
+  return cookie[cookieName];
+};
+
+export const setSessionId = (cookieName: string, domain?: string): void => {
+  const sessionId = getSessionId(cookieName) || crypto.randomUUID();
+  const baseCookie = `${cookieName}=${sessionId}; Max-Age=1800; path=/; secure`;
+  const cookie = domain ? baseCookie + `; domain=${domain}` : baseCookie;
+
+  document.cookie = cookie;
+};
 
 export const useEventTracker = () => {
-  const telemetry = useRecoilValue(telemetryState);
   const [createEventMutation] = useTrackMutation();
 
   return useCallback(
-    (eventType: string, eventData: EventData) => {
-      if (telemetry.enabled) {
-        createEventMutation({
-          variables: {
-            type: eventType,
-            data: eventData,
-          },
-        });
-      }
+    (eventType: string, sessionId: string, eventData: EventData) => {
+      createEventMutation({
+        variables: {
+          type: eventType,
```

Comment:
why not copy the payload from the example? Here we send pageview event from the frontend but you'll want to track other types of event